### ISSUE TITLE
list_account_subscriptions example incorrect

### ISSUE THREAD

**Issue by noctivityinc** [recurly/recurly-client-ruby#675]
The list_account_subscriptions example says this:

```
params = {
  limit: 200
}
subscriptions = @client.list_account_subscriptions(
  account_id: account_id,
  params: params
)
```

Which is incorrect and caused a validation error.  You need to send params directly, like this:

```
subscriptions = @client.list_account_subscriptions(
  account_id: account_id,
  limit: 200
)
```

Please update the README and docs.

**Comment by douglasmiller** [comment]
I saw in another issue that you're using the `3.18.1` version of the client. We released the `4.0.0` version of the client at the beginning of the month which changes this behavior.

The `master` branch is for the current `4.0.0` version of the client which pairs with the `2021-02-25` version of Recurly API. The `3.x` client pairs with the `2019-10-10` version of Recurly API and it's source can be found on the [v3-v2019-10-10](https://github.com/recurly/recurly-client-ruby/tree/v3-v2019-10-10) branch of the repository. The documentation for the `2019-10-10` API can be found on the [developer hub](https://developers.recurly.com/api/v2019-10-10/index.html) as well as a listing of [available documentation versions](https://developers.recurly.com/api/index.html).

This change was introduced with pull request https://github.com/recurly/recurly-client-ruby/pull/619 and is included in the breaking changes section of the [CHANGELOG](https://github.com/recurly/recurly-client-ruby/blob/master/CHANGELOG.md#400-2021-03-01) for the 4.x release.

I apologize for the confusion. We will look into adding more visibility to these differences to the README.

**Comment by noctivityinc** [comment]
@douglasmiller you may want to update the getting started, as this is right at the top:

```
gem 'recurly', '~> 3.18'
```

**Comment by douglasmiller** [comment]
Where are you seeing that? The [GETTING_STARTED on master](https://github.com/recurly/recurly-client-ruby/blob/master/GETTING_STARTED.md) has 4.0. If there's something off, I'd like to fix it.

**Comment by noctivityinc** [comment]
Sorry @douglasmiller , this may be ANOTHER 3.0 issue.  I upgraded the gem to 4.0 as you requested and this is the method:

```
 params = {
      limit: 200
    }

    @subscriptions = RECURLY_CLIENT.list_account_subscriptions(
      account_id: account_id
      params: params
    )
```

and I get this:

```
wrong number of arguments (given 1, expected 0)
```

**Comment by noctivityinc** [comment]
Relevant trace...

```
Application Trace | Framework Trace | Full Trace
recurly (4.0.0) lib/recurly/client.rb:299:in `validate_path_parameters!'
recurly (4.0.0) lib/recurly/client.rb:319:in `interpolate_path'
recurly (4.0.0) lib/recurly/client/operations.rb:1233:in `list_account_subscriptions'
app/controllers/subscriptions_controller.rb:13:in `index'
```

**Comment by noctivityinc** [comment]
@douglasmiller any updates on this one?  I see it's still closed even though it appears broken.

**Comment by douglasmiller** [comment]
The ruby 3.0.0 fixes were not included in the 4.0.0 client library release. They were merged up to the 4.x client branch in https://github.com/recurly/recurly-client-ruby/pull/671 which we will be releasing soon.

**Comment by noctivityinc** [comment]
ahh - that explains what I just found.

When I use this:
```gem 'recurly', '~> 4.0'``` 

I get errors but when I reference the github branch directly:
```
gem 'recurly', '~> 4.0', github: 'recurly/recurly-client-ruby'
```

Then it works properly.  Count still doesn't work, which I know you are fixing, but good to know about the other situation.  You guys really need to update your README for these situations.  I wasted the day.

**Comment by douglasmiller** [comment]
@noctivityinc I just published the 4.0.1 version of the client which fixes this ruby 3.x issue with the `Pager#count` functionality.
